programs/session-manager/src/token/revoke.rs

```diff
@@ -0,0 +1,52 @@
+use crate::error::SessionManagerError;
+use crate::{CloseSession, SESSION_SETTER_SEED};
+use anchor_lang::prelude::*;
+use anchor_lang::solana_program::program_option::COption;
+use anchor_spl::associated_token::get_associated_token_address;
+use anchor_spl::token::{revoke, Revoke, TokenAccount};
+
+impl<'info> CloseSession<'info> {
+    /// Revoke token accounts from the session key.
+    /// When closing a session, the session account is returned to the system program.
+    /// We need to revoke all token delegations to the session key, otherwise the session key could still have power to spend tokens from the user accounts even if the session had expired or been revoked.
+    pub fn revoke_tokens(
+        &self,
+        accounts: &[AccountInfo<'info>],
```

Comment:
`accounts` and `mints_to_revoke` could probably be zipped together into a single type, like

```
pub struct RevokeTokenAccount<'info> {
    account: Account<'info, TokenAccount>,
    mint: &Pubkey,
}
```

then you wouldn't need to do the gte check or the TokenAccount deserialization here

Reply:
refactored